### ISSUE TITLE
Will the dataset be shuffled randomly when fine-tuning on a custom dataset?

### ISSUE THREAD

**Issue by panovr** [facebook/fb.resnet.torch#159]
I prepare my custom dataset following the guide like:
```
train/<label1>/<image.jpg>
train/<label2>/<image.jpg>
val/<label1>/<image.jpg>
val/<label2>/<image.jpg>
```
I just want to ask that will the dataset be shuffled randomly when fine-tuning?

**Comment by aabobakr** [comment]
Yes. Data are shuffled every epoch. In the dataloader.lua this, line generates the random permutation every epoch `local perm = torch.randperm(size)`

**Comment by panovr** [comment]
That's it, thanks!
